File: src/skills/petition-drafter/piece-types.js
/**
 * Catálogo de tipos de peça e seus requisitos formais.
 *
 * Estrutura inspirada no catálogo do projeto Sistema de Petições (MIT), com o
 * escopo ampliado do previdenciário para as áreas em que o produto opera e com
 * o checklist formal explícito por tipo — é ele que o `enrich` roda em código
 * sobre a peça gerada, antes de liberar a exportação.
 */

export const TIPOS_PECA = {
    "peticao-inicial": {
        nome: "Petição Inicial",
        grupo: "Postulatórias",
        rito: "judicial",
        estrutura: [
            "endereçamento",
            "qualificação das partes",
            "dos fatos",
            "do direito",
            "da tutela de urgência (quando cabível)",
            "dos pedidos",
            "das provas",
            "do valor da causa",
            "fecho e assinatura"
        ],
        checklist: [
            { requisito: "Juízo a que é dirigida", fundamento: "art. 319, I, CPC" },
            { requisito: "Qualificação completa das partes", fundamento: "art. 319, II, CPC" },
            { requisito: "Fato e fundamentos jurídicos do pedido", fundamento: "art. 319, III, CPC" },
            { requisito: "Pedido com suas especificações (certo e determinado)", fundamento: "arts. 319, IV e 322, CPC" },
            { requisito: "Valor da causa", fundamento: "arts. 319, V e 291, CPC" },
            { requisito: "Provas com que pretende demonstrar a verdade dos fatos", fundamento: "art. 319, VI, CPC" },
            { requisito: "Opção pela realização ou não de audiência de conciliação", fundamento: "art. 319, VII, CPC" }
        ]
    },

    contestacao: {
        nome: "Contestação",
        grupo: "Respostas",
        rito: "judicial",
        estrutura: [
            "endereçamento",
            "preliminares",
            "prejudiciais de mérito",
            "impugnação especificada dos fatos",
            "do mérito",
            "dos pedidos",
            "das provas",
            "fecho e assinatura"
        ],
        checklist: [
            { requisito: "Toda a matéria de defesa concentrada na peça", fundamento: "art. 336, CPC" },
            { requisito: "Preliminares antes do mérito", fundamento: "art. 337, CPC" },
            { requisito: "Impugnação especificada de cada fato — o que não se impugna presume-se verdadeiro", fundamento: "art. 341, CPC" },
            { requisito: "Prejudiciais de prescrição e decadência, se houver", fundamento: "art. 487, II, CPC" },
            { requisito: "Pedido de improcedência", fundamento: "art. 336, CPC" },
            { requisito: "Protesto por provas", fundamento: "art. 336, CPC" }
        ]
    },

    replica: {
        nome: "Réplica / Impugnação à Contestação",
        grupo: "Respostas",
        rito: "judicial",
        estrutura: ["endereçamento", "enfrentamento das preliminares", "réplica ao mérito", "reafirmação dos pedidos", "das provas"],
        checklist: [
            { requisito: "Enfrentamento de cada preliminar suscitada", fundamento: "arts. 350 e 351, CPC" },
            { requisito: "Manifestação sobre documentos juntados com a contestação", fundamento: "art. 437, §1º, CPC" },
            { requisito: "Reafirmação dos pedidos da inicial", fundamento: "art. 329, CPC" }
        ]
    },

    "apelacao": {
        nome: "Apelação Cível",
        grupo: "Recursos",
        rito: "judicial",
        estrutura: ["endereçamento ao juízo a quo", "razões de apelação", "tempestividade", "dos fatos", "do direito — capítulos impugnados", "prequestionamento", "dos pedidos de reforma", "preparo"],
        checklist: [
            { requisito: "Nome e qualificação das partes", fundamento: "art. 1.010, I, CPC" },
            { requisito: "Exposição do fato e do direito", fundamento: "art. 1.010, II, CPC" },
            { requisito: "Razões do pedido de reforma ou decretação de nulidade", fundamento: "art. 1.010, III, CPC" },
            { requisito: "Pedido de nova decisão", fundamento: "art. 1.010, IV, CPC" },
            { requisito: "Demonstração de tempestividade (15 dias úteis)", fundamento: "arts. 1.003, §5º e 219, CPC" },
            { requisito: "Comprovante de preparo ou pedido de gratuidade", fundamento: "art. 1.007, CPC" },
            { requisito: "Impugnação específica dos fundamentos da sentença", fundamento: "Súmula 182 do STJ, por analogia" }
        ]
    },

    "agravo-instrumento": {
        nome: "Agravo de Instrumento",
        grupo: "Recursos",
        rito: "judicial",
        estrutura: ["endereçamento ao tribunal", "cabimento", "tempestividade", "dos fatos", "do direito", "do efeito suspensivo ou tutela recursal", "dos pedidos", "peças obrigatórias"],
        checklist: [
            { requisito: "Hipótese de cabimento do rol do art. 1.015", fundamento: "art. 1.015, CPC" },
            { requisito: "Nomes e qualificação completa das partes", fundamento: "art. 1.016, I, CPC" },
            { requisito: "Exposição do fato e do direito e razões do pedido de reforma", fundamento: "art. 1.016, II e III, CPC" },
            { requisito: "Nome e endereço dos advogados", fundamento: "art. 1.016, IV, CPC" },
            { requisito: "Peças obrigatórias do art. 1.017", fundamento: "art. 1.017, I, CPC" },
            { requisito: "Tempestividade (15 dias úteis)", fundamento: "art. 1.003, §5º, CPC" }
        ]
    },

    "embargos-declaracao": {
        nome: "Embargos de Declaração",
        grupo: "Recursos",
        rito: "judicial",
        estrutura: ["endereçamento", "tempestividade", "do vício apontado", "do prequestionamento", "do pedido"],
        checklist: [
            { requisito: "Indicação precisa de omissão, contradição, obscuridade ou erro material", fundamento: "art. 1.022, CPC" },
            { requisito: "Tempestividade (5 dias úteis)", fundamento: "art. 1.023, CPC" },
            { requisito: "Pedido de integração ou correção, não de rediscussão do mérito", fundamento: "art. 1.022, CPC" }
        ]
    },

    contrarrazoes: {
        nome: "Contrarrazões de Recurso",
        grupo: "Recursos",
        rito: "judicial",
        estrutura: ["endereçamento", "preliminares de não conhecimento", "do mérito recursal", "do pedido de manutenção"],
        checklist: [
            { requisito: "Preliminares de admissibilidade, se houver", fundamento: "art. 1.010, §1º, CPC" },
            { requisito: "Enfrentamento de cada fundamento do recurso", fundamento: "art. 1.010, §1º, CPC" },
            { requisito: "Pedido de manutenção da decisão recorrida", fundamento: "art. 1.010, §1º, CPC" }
        ]
    },

    "reclamacao-trabalhista": {
        nome: "Reclamação Trabalhista",
        grupo: "Postulatórias",
        rito: "trabalhista",
        estrutura: ["endereçamento à Vara do Trabalho", "qualificação", "dos fatos", "do direito", "dos pedidos com valores", "das provas", "do valor da causa"],
        checklist: [
            { requisito: "Designação do juízo, qualificação das partes, breve exposição dos fatos e pedido", fundamento: "art. 840, §1º, CLT" },
            { requisito: "Pedidos certos, determinados e com indicação de valor", fundamento: "art. 840, §1º, CLT" },
            { requisito: "Data e assinatura", fundamento: "art. 840, §1º, CLT" }
        ]
    },

    "mandado-seguranca": {
        nome: "Mandado de Segurança",
        grupo: "Ações constitucionais",
        rito: "judicial",
        estrutura: ["endereçamento", "qualificação do impetrante e da autoridade coatora", "dos fatos", "do direito líquido e certo", "da liminar", "dos pedidos"],
        checklist: [
            { requisito: "Indicação da autoridade coatora e da pessoa jurídica a que se vincula", fundamento: "art. 6º, Lei 12.016/2009" },
            { requisito: "Prova pré-constituída do direito líquido e certo", fundamento: "art. 6º, Lei 12.016/2009" },
            { requisito: "Prazo decadencial de 120 dias", fundamento: "art. 23, Lei 12.016/2009" }
        ]
    },

    "peticao-simples": {
        nome: "Petição Simples / Intermediária",
        grupo: "Intermediárias",
        rito: "judicial",
        estrutura: ["endereçamento", "referência ao processo", "da finalidade", "do pedido"],
        checklist: [
            { requisito: "Identificação do processo e das partes", fundamento: "prática forense" },
            { requisito: "Finalidade objetiva e pedido determinado", fundamento: "art. 322, CPC" }
        ]
    },

    "notificacao-extrajudicial": {
        nome: "Notificação Extrajudicial",
        grupo: "Extrajudicial",
        rito: "extrajudicial",
        estrutura: ["destinatário", "do fato", "da exigência", "do prazo", "da consequência do descumprimento"],
        checklist: [
            { requisito: "Identificação inequívoca do notificante e do notificado", fundamento: "prática" },
            { requisito: "Exigência clara com prazo determinado", fundamento: "art. 397, parágrafo único, CC" },
            { requisito: "Consequência jurídica do descumprimento", fundamento: "prática" }
        ]
    },

    parecer: {
        nome: "Parecer Jurídico",
        grupo: "Consultivo",
        rito: "consultivo",
        estrutura: ["da consulta", "dos fatos", "da análise jurídica", "da conclusão", "das ressalvas"],
        checklist: [
            { requisito: "Questão posta de forma objetiva", fundamento: "boa técnica" },
            { requisito: "Premissas fáticas declaradas", fundamento: "boa técnica" },
            { requisito: "Conclusão objetiva com grau de segurança declarado", fundamento: "boa técnica" },
            { requisito: "Ressalva sobre dados não verificados", fundamento: "boa técnica" }
        ]
    }
};

export const LISTA_TIPOS = Object.entries(TIPOS_PECA).map(([id, t]) => ({
    id,
    nome: t.nome,
    grupo: t.grupo,
    rito: t.rito
}));

export function getTipoPeca(id) {
    return TIPOS_PECA[id] || null;
}

/**
 * Agrupa os tipos por grupo, para os selects da interface.
 */
export function tiposPorGrupo() {
    return LISTA_TIPOS.reduce((acc, tipo) => {
        (acc[tipo.grupo] ||= []).push(tipo);
        return acc;
    }, {});
}

File: referencias/sistema-de-peticoes/core/piece_types.py
"""Catálogo de tipos de peças alinhado ao prompt jurídico versionado."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class PieceType:
    id: str
    nome: str
    grupo: str
    profile_id: str
    exige_revisao: str


PIECE_TYPES: tuple[PieceType, ...] = (
    PieceType("auxilio-incapacidade-temporaria", "Petição Inicial — Auxílio por Incapacidade Temporária", "Benefícios por incapacidade", "judicial-inicial-jef", "Base clínica, DII, profissão habitual e documentos médicos."),
    PieceType("aposentadoria-incapacidade-permanente", "Petição Inicial — Aposentadoria por Incapacidade Permanente", "Benefícios por incapacidade", "judicial-inicial-jef", "Base clínica, incapacidade total/permanente e condições pessoais."),
    PieceType("aposentadoria-invalidez-acidentaria", "Petição Inicial — Aposentadoria por Invalidez Acidentária (B-92)", "Benefícios por incapacidade", "judicial-inicial-estadual", "Nexo ocupacional, CAT e competência estadual."),
    PieceType("auxilio-acidente", "Petição Inicial — Auxílio-Acidente (B-36)", "Benefícios por incapacidade", "judicial-inicial-jef", "Consolidação das lesões e redução da capacidade laboral."),
    PieceType("restabelecimento-beneficio-incapacidade", "Petição Inicial — Restabelecimento de Benefício por Incapacidade Cessado", "Benefícios por incapacidade", "judicial-inicial-jef", "DCB, laudos contemporâneos e manutenção da incapacidade."),
    PieceType("aposentadoria-idade-urbana", "Petição Inicial — Aposentadoria por Idade Urbana", "Aposentadorias", "judicial-inicial-jef", "Carência, idade, CNIS e DER."),
    PieceType("aposentadoria-idade-rural", "Petição Inicial — Aposentadoria por Idade Rural", "Aposentadorias", "judicial-inicial-jef", "Prova rural, início de prova material e testemunhas."),
    PieceType("aposentadoria-hibrida", "Petição Inicial — Aposentadoria Híbrida", "Aposentadorias", "judicial-inicial-jef", "Períodos rurais/urbanos e cálculo de carência."),
    PieceType("aposentadoria-tempo-contribuicao", "Petição Inicial — Aposentadoria por Tempo de Contribuição", "Aposentadorias", "judicial-inicial-jef", "Direito adquirido ou regra de transição após EC 103/2019."),
    PieceType("aposentadoria-especial", "Petição Inicial — Aposentadoria Especial", "Aposentadorias", "judicial-inicial-jef", "PPP, LTCAT, agentes nocivos e conversão quando cabível."),
    PieceType("aposentadoria-pcd-tempo", "Petição Inicial — Aposentadoria por Tempo — Pessoa com Deficiência", "Aposentadorias", "judicial-inicial-jef", "Avaliação biopsicossocial e grau de deficiência."),
    PieceType("revisao-aposentadoria", "Petição Inicial — Revisão de Aposentadoria", "Aposentadorias", "judicial-inicial-jef", "RMI, DIB, CNIS, carta de concessão e tese revisional."),
    PieceType("pensao-por-morte", "Petição Inicial — Pensão por Morte", "Outros benefícios previdenciários", "judicial-inicial-jef", "Data do óbito, qualidade de segurado e dependência."),
    PieceType("salario-maternidade", "Petição Inicial — Salário-Maternidade", "Outros benefícios previdenciários", "judicial-inicial-jef", "Nascimento/adoção, qualidade de segurada e carência quando aplicável."),
    PieceType("reconhecimento-tempo-contribuicao", "Petição Inicial — Reconhecimento de Tempo de Contribuição", "Outros benefícios previdenciários", "judicial-inicial-jef", "Vínculos, CTPS, CNIS e provas complementares."),
    PieceType("bpc-deficiencia-judicial", "Petição Inicial — BPC/LOAS — Pessoa com Deficiência", "BPC/LOAS judicial", "judicial-inicial-jef", "Impedimento de longo prazo, CadÚnico e miserabilidade."),
    PieceType("bpc-idoso-judicial", "Petição Inicial — BPC/LOAS — Idoso", "BPC/LOAS judicial", "judicial-inicial-jef", "Idade mínima, composição familiar e renda."),
    PieceType("ms-bpc-mora", "Mandado de Segurança — BPC/LOAS por Mora Administrativa", "BPC/LOAS judicial", "judicial-inicial-jef", "Distinguir mora continuada de indeferimento expresso."),
    PieceType("requerimento-bpc-idoso", "Requerimento Administrativo — BPC/LOAS — Idoso", "BPC/LOAS administrativo", "administrativo-inss", "CadÚnico, composição familiar e renda."),
    PieceType("requerimento-bpc-deficiencia", "Requerimento Administrativo — BPC/LOAS — Pessoa com Deficiência", "BPC/LOAS administrativo", "administrativo-inss", "Base clínica e avaliação biopsicossocial."),
    PieceType("recurso-bpc", "Recurso Administrativo — BPC/LOAS", "BPC/LOAS administrativo", "administrativo-inss", "Decisão administrativa, fundamentos de impugnação e documentos."),
    PieceType("requerimento-inss-geral", "Requerimento Administrativo ao INSS — Geral", "Administrativo INSS", "administrativo-inss", "Pedido objetivo, documentos e identificação do requerente."),
    PieceType("recurso-crps", "Recurso Administrativo ao INSS / CRPS", "Administrativo INSS", "administrativo-inss", "Decisão, prazo, pontos impugnados e provas."),
    PieceType("cumprimento-exigencia", "Cumprimento de Exigência Administrativa", "Administrativo INSS", "administrativo-inss", "Texto integral da exigência e resposta item a item."),
    PieceType("pedido-prioridade", "Pedido de Prioridade de Tramitação", "Administrativo INSS", "administrativo-inss", "Idade, doença grave ou fundamento legal da prioridade."),
    PieceType("mandado-seguranca-previdenciario", "Mandado de Segurança Previdenciário", "Ações especiais", "judicial-inicial-jef", "Ato coator, autoridade, prazo decadencial e prova pré-constituída."),
    PieceType("tutela-antecedente", "Tutela Antecipada em Caráter Antecedente", "Ações especiais", "judicial-inicial-estadual", "Probabilidade do direito, perigo de dano e pedido final futuro."),
    PieceType("recurso-inominado", "Recurso Inominado (JEF)", "Recursos e impugnações", "forense-basico", "Sentença, prazo, fundamentos recursais e pedidos."),
    PieceType("apelacao-civel", "Apelação Cível", "Recursos e impugnações", "forense-basico", "Sentença, capítulos impugnados e preparo quando cabível."),
    PieceType("agravo-instrumento", "Agravo de Instrumento", "Recursos e impugnações", "forense-basico", "Decisão agravada, urgência e peças obrigatórias."),
    PieceType("embargos-declaracao", "Embargos de Declaração", "Recursos e impugnações", "forense-basico", "Omissão, contradição, obscuridade ou erro material."),
    PieceType("contrarrazoes", "Contrarrazões de Recurso", "Recursos e impugnações", "forense-basico", "Recurso adverso, preliminares e manutenção da decisão."),
    PieceType("cumprimento-sentenca-implantacao", "Cumprimento de Sentença — Implantação do Benefício", "Cumprimento de sentença", "forense-basico", "Título judicial, trânsito/fase e obrigação de fazer."),
    PieceType("cumprimento-sentenca-rpv", "Cumprimento de Sentença — Expedição de RPV", "Cumprimento de sentença", "forense-basico", "Cálculos, teto de RPV e dados bancários quando necessários."),
    PieceType("impugnacao-calculos", "Impugnação aos Cálculos", "Cumprimento de sentença", "forense-basico", "Memória de cálculo, divergência objetiva e documentos."),
    PieceType("juntada-documentos", "Petição de Juntada de Documentos", "Petições intermediárias", "forense-basico", "Processo, documentos e finalidade da juntada."),
    PieceType("replica-contestacao", "Réplica / Impugnação à Contestação", "Petições intermediárias", "forense-basico", "Contestação, preliminares e pontos controvertidos."),
    PieceType("manifestacao-documentos", "Manifestação sobre Documentos Novos", "Petições intermediárias", "forense-basico", "Documentos, pertinência e contraditório."),
    PieceType("peticao-simples", "Petição Simples / Outro", "Petições intermediárias", "forense-basico", "Finalidade objetiva e contexto processual."),
    PieceType("inventario-extrajudicial", "Inventário Extrajudicial", "Sucessório e extrajudicial", "extrajudicial-tabelionato", "Herdeiros, bens, ITCMD, testamento e incapazes."),
    PieceType("alvara-judicial", "Alvará Judicial", "Sucessório e extrajudicial", "judicial-inicial-estadual", "Cabimento, valores/bens e interessados."),
    PieceType("usucapiao", "Usucapião", "Cível e sucessório", "judicial-inicial-estadual", "Posse, tempo, imóvel, confrontantes e documentos."),
    PieceType("revisao-auxilio-incapacidade-temporaria", "Petição Inicial — Revisão de Auxílio por Incapacidade Temporária", "Benefícios por incapacidade", "judicial-inicial-jef", "DIB, RMI, período devido e base clínica da revisão."),
    PieceType("revisao-auxilio-acidente", "Petição Inicial — Revisão de Auxílio-Acidente", "Benefícios por incapacidade", "judicial-inicial-jef", "Critério de cálculo, sequelas e documentação médica."),
    PieceType("aposentadoria-pcd-idade", "Petição Inicial — Aposentadoria por Idade — Pessoa com Deficiência", "Aposentadorias", "judicial-inicial-jef", "Idade, deficiência, carência e avaliação biopsicossocial."),
    PieceType("reconhecimento-atividade-especial", "Petição Inicial — Reconhecimento de Atividade Especial", "Outros benefícios previdenciários", "judicial-inicial-jef", "PPP, LTCAT, agentes nocivos e enquadramento por período."),
    PieceType("auxilio-reclusao", "Petição Inicial — Auxílio-Reclusão", "Outros benefícios previdenciários", "judicial-inicial-jef", "Qualidade de segurado, baixa renda, dependência e certidão de reclusão."),
    PieceType("revisao-pensao-por-morte", "Petição Inicial — Revisão de Pensão por Morte", "Outros benefícios previdenciários", "judicial-inicial-jef", "Óbito, dependência, base de cálculo e tese revisional."),
    PieceType("bpc-revisao-restabelecimento", "Petição Inicial — Revisão / Restabelecimento de BPC/LOAS", "BPC/LOAS judicial", "judicial-inicial-jef", "Decisão de cessação/revisão, composição familiar e prova social atual."),
    PieceType("recurso-ordinario-crps", "Recurso Ordinário ao CRPS", "Administrativo INSS", "administrativo-inss", "Decisão administrativa, prazo recursal e fundamentos de reforma."),
    PieceType("recurso-especial-crps", "Recurso Especial ao CRPS", "Administrativo INSS", "administrativo-inss", "Divergência, admissibilidade e decisão recorrida."),
    PieceType("pedido-reconsideracao-administrativa", "Pedido de Reconsideração Administrativa", "Administrativo INSS", "administrativo-inss", "Decisão, fato novo ou erro administrativo objetivo."),
    PieceType("ctc", "Requerimento — Certidão de Tempo de Contribuição (CTC)", "Serviços administrativos INSS", "administrativo-inss", "Regime de destino, períodos e impedimentos de contagem recíproca."),
    PieceType("copia-processo-administrativo", "Requerimento — Cópia Integral do Processo Administrativo", "Serviços administrativos INSS", "administrativo-inss", "NB/protocolo, identificação do interessado e finalidade."),
    PieceType("retificacao-cnis", "Requerimento — Retificação / Atualização de CNIS", "Serviços administrativos INSS", "administrativo-inss", "Vínculos, remunerações, provas e período a corrigir."),
    PieceType("justificacao-administrativa", "Justificação Administrativa", "Serviços administrativos INSS", "administrativo-inss", "Fatos a provar, testemunhas e documentos mínimos."),
    PieceType("acerto-vinculos-remuneracoes", "Requerimento — Acerto de Vínculos e Remunerações no CNIS", "Serviços administrativos INSS", "administrativo-inss", "Vínculo, remuneração, competência e prova documental."),
    PieceType("regularizacao-representante", "Requerimento — Regularização de Representante / Procurador", "Serviços administrativos INSS", "administrativo-inss", "Documento de representação, poderes e dados do procurador."),
    PieceType("agravo-interno", "Agravo Interno", "Recursos e impugnações", "forense-basico", "Decisão monocrática, prazo e fundamentos de reforma."),
    PieceType("pedilef-tnu", "Pedido de Uniformização de Interpretação de Lei (PEDILEF/TNU)", "Recursos e impugnações", "forense-basico", "Divergência entre turmas, questão de direito material e admissibilidade."),
    PieceType("recurso-especial-stj", "Recurso Especial (STJ)", "Recursos e impugnações", "forense-basico", "Violação de lei federal, divergência e juízo de admissibilidade."),
    PieceType("recurso-extraordinario-stf", "Recurso Extraordinário (STF)", "Recursos e impugnações", "forense-basico", "Questão constitucional, repercussão geral e admissibilidade."),
    PieceType("juizo-retratacao", "Juízo de Retratação", "Recursos e impugnações", "forense-basico", "Tese vinculante, recurso repetitivo ou precedente aplicável."),
    PieceType("cumprimento-sentenca-precatorio", "Cumprimento de Sentença — Expedição de Precatório", "Cumprimento de sentença", "forense-basico", "Valor acima do teto de RPV, cálculos e dados necessários."),
    PieceType("cumprimento-sentenca-astreintes", "Cumprimento de Sentença — Astreintes por Descumprimento", "Cumprimento de sentença", "forense-basico", "Ordem judicial, prazo, descumprimento e cálculo da multa."),
    PieceType("impugnacao-cumprimento-sentenca", "Impugnação ao Cumprimento de Sentença", "Cumprimento de sentença", "forense-basico", "Excesso, inexigibilidade, nulidade ou tese defensiva cabível."),
    PieceType("habilitacao-sucessoria-processo", "Habilitação Sucessória no Curso do Processo", "Cumprimento de sentença", "forense-basico", "Óbito, sucessores, documentos e regularização processual."),
    PieceType("tutela-urgencia-incidental", "Petição de Tutela de Urgência / Antecipada", "Petições intermediárias", "forense-basico", "Probabilidade, perigo de dano e fase processual."),
    PieceType("impugnacao-laudo-pericial", "Impugnação ao Laudo Pericial", "Petições intermediárias", "forense-basico", "Laudo, contradições, quesitos e pedido de esclarecimentos."),
    PieceType("quesitos-periciais", "Apresentação de Quesitos Periciais", "Petições intermediárias", "forense-basico", "Objeto da perícia, quesitos técnicos e documentos de apoio."),
    PieceType("especificacao-provas", "Especificação de Provas", "Petições intermediárias", "forense-basico", "Fatos controvertidos, provas pretendidas e pertinência."),
    PieceType("sobrepartilha-extrajudicial", "Sobrepartilha Extrajudicial", "Sucessório e extrajudicial", "extrajudicial-tabelionato", "Bens sobrevindos, consenso e requisitos notariais."),
    PieceType("cessao-direitos-hereditarios", "Cessão de Direitos Hereditários", "Sucessório e extrajudicial", "extrajudicial-tabelionato", "Forma pública, cedente, cessionário, objeto e anuências."),
    PieceType("renuncia-heranca", "Renúncia à Herança / Repúdio", "Sucessório e extrajudicial", "extrajudicial-tabelionato", "Forma pública ou termo judicial e análise de credores."),
    PieceType("inventario-judicial", "Inventário Judicial — Petição Inicial", "Cível e sucessório", "judicial-inicial-estadual", "Óbito, bens, herdeiros, ITCMD, testamento e incapazes."),
    PieceType("arrolamento-simples", "Arrolamento Simples — Petição Inicial", "Cível e sucessório", "judicial-inicial-estadual", "Óbito, bens, herdeiros e adequação do rito."),
    PieceType("arrolamento-sumario", "Arrolamento Sumário — Petição Inicial", "Cível e sucessório", "judicial-inicial-estadual", "Consenso, herdeiros capazes, partilha e adequação do rito."),
    PieceType("sobrepartilha-judicial", "Sobrepartilha — Petição Inicial", "Cível e sucessório", "judicial-inicial-estadual", "Bens descobertos após partilha e legitimidade."),
    PieceType("formal-partilha", "Formal de Partilha / Carta de Adjudicação", "Cível e sucessório", "forense-basico", "Decisão homologatória, trânsito e dados dos bens/herdeiros."),
    PieceType("habilitacao-herdeiros", "Habilitação de Herdeiros", "Cível e sucessório", "forense-basico", "Óbito, sucessores e prova documental."),
    PieceType("primeiras-declaracoes", "Primeiras Declarações do Inventariante", "Cível e sucessório", "forense-basico", "Inventariante, herdeiros, bens, dívidas e plano inicial."),
    PieceType("ultimas-declaracoes", "Últimas Declarações do Inventariante", "Cível e sucessório", "forense-basico", "Atualização de bens, dívidas, partilha e concordâncias."),
    PieceType("nomeacao-inventariante", "Nomeação / Substituição / Remoção de Inventariante", "Cível e sucessório", "forense-basico", "Legitimidade, motivo e documentos do inventário."),
    PieceType("procuracao-ad-judicia", "Procuração Ad Judicia", "Instrumentos de mandato", "instrumento-mandato", "Outorgante, outorgado, poderes, foro, data e assinatura."),
    PieceType("procuracao-administrativa-inss", "Procuração Administrativa Previdenciária / INSS", "Instrumentos de mandato", "instrumento-mandato", "Outorgante, procurador, poderes administrativos e validade."),
    PieceType("procuracao-ad-judicia-et-extra", "Procuração Ad Judicia et Extra", "Instrumentos de mandato", "instrumento-mandato", "Poderes judiciais e extrajudiciais devem ser expressos e revisados."),
    PieceType("substabelecimento-com-reserva", "Substabelecimento com Reserva de Poderes", "Instrumentos de mandato", "instrumento-mandato", "Procuração originária, substabelecente, substabelecido e reserva expressa."),
    PieceType("substabelecimento-sem-reserva", "Substabelecimento sem Reserva de Poderes", "Instrumentos de mandato", "instrumento-mandato", "Procuração originária, ciência do cliente e transferência integral de poderes."),
    PieceType("declaracao-hipossuficiencia", "Declaração de Hipossuficiência", "Instrumentos e declarações", "instrumento-mandato", "Declarante, situação econômica, data e assinatura."),
    PieceType("declaracao-residencia", "Declaração de Residência", "Instrumentos e declarações", "instrumento-mandato", "Declarante, endereço, finalidade e assinatura."),
    PieceType("declaracao-atividade-rural", "Declaração de Atividade Rural", "Instrumentos e declarações", "instrumento-mandato", "Período, atividade, local, testemunhas e documentos."),
)


def list_piece_types() -> list[PieceType]:
    return sorted(PIECE_TYPES, key=lambda item: (item.grupo, item.nome))


def get_piece_type(piece_type_id: str | None) -> PieceType | None:
    if not piece_type_id:
        return None
    for piece_type in PIECE_TYPES:
        if piece_type.id == piece_type_id:
            return piece_type
    raise ValueError(f"tipo de peça desconhecido: {piece_type_id}")

